gengokit/gengokit.go

```diff
@@ -125,31 +223,39 @@ func (g *generator) updateServerMethods(svcHandler io.Reader, svcFuncs []string)
 	astMod.RemoveFunctionsExecpt(svcFuncs)
 	astMod.RemoveInterface("Service")
 
-	// Index the handler functions, apply handler template for all function in service definition that are not defined in handler
+	// Index the handler functions, apply handler template for all function in
+	// service definition that are not defined in handler
 	currentFuncs := astMod.IndexFunctions()
 	code := astMod.Buffer()
 
-	err = g.applyTemplateForMissingMeths(svcMethodsTemplPath, currentFuncs, code)
+	trimmedTe, err := trimTemplateExecutorServiceFuncs(*te, currentFuncs)
+
+	out, err := applyTemplateFromPath(svcMethodsTemplPath, trimmedTe)
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to apply service methods template")
 	}
 
+	code.ReadFrom(out)
+
 	// Insert updated Service interface
-	outBuf, err := applyTemplateFromPath(svcInterfaceTemplPath, g.templateExec)
+	out, err = applyTemplateFromPath(svcInterfaceTemplPath, te)
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to apply service interface template")
 	}
 
-	code.Write(outBuf.Bytes())
+	code.ReadFrom(out)
 
 	return code, nil
 }
 
-// updateClientMethods will update the functions within an existing
-// client_handler.go file so that it contains exactly the fucntions passed in
-// svcFuncs, no more, no less.
-func (g *generator) updateClientMethods(clientHandler io.Reader, svcFuncs []string) (outCode *bytes.Buffer, err error) {
+// updateClientMethods accepts NAME-service/handlers/client/client_handler.go
+// as an io.Reader and modifies it to contains only functions returned by
+// serviceFunctionNames.
+func updateClientMethods(clientHandler io.Reader, te *templateExecutor) (outCode io.Reader, err error) {
+	svcFuncs := serviceFunctionsNames(te.Service.Methods)
+
```

Comment:
Something about declaring constants inside a function seems weird. I imagine there's no penalty since the compiler and move them out of the actual function execution, but I typically expect constants declared at a package level, like,

```
const (
   myFirstConst = "My first constant!"
   mySecondConst = "Hey gang, this is my second const."
   myFourthConst = "What happened to the third constant?"
   myFifthConst = "We don't talk about the third constant."
)
```
I'm sort of torn, because having them declared right in the function is very readable since it maintains locality, but traditionally, if something is never going to change, it's not declared inside a function body.